IBM/CAST
# Bills db migration 18.0 and more

# Purpose
Migrate CSM_DB version from 17.0 to 18.0

Added fields within the CSM DB tables which are a part of the new migration upgrade.

| csm_allocation table | Details
|----------------------|---------------------------------
| core_blink           | boolean     not null,
| COMMENT ON COLUMN    | csm_allocation.core_blink is 'flag indicating whether or not to run a blink operation on allocation cores.';

| csm_allocation_history | Details
|----------------------|---------------------------------
| core_blink           | boolean     not null
| COMMENT ON COLUMN    |csm_allocation_history.core_blink is 'flag indicating whether or not to run a blink operation on allocation cores.';

| csm_switch_inventory | Details
|----------------------|---------------------------------
| type                 | text,
| fw_version           | text,
| COMMENT ON COLUMN    | csm_switch_inventory.type is 'The category of this piece of hardware inventory. For example: "FAN", "PS", "SYSTEM", or "MGMT".';
| COMMENT ON COLUMN    | csm_switch_inventory.fw_version is 'The firmware version on this piece of inventory.';

| csm_switch_inventory_history | Details
|----------------------|---------------------------------
| archive_history_time | timestamp,
| type                 | text,
| fw_version           | text
| COMMENT ON COLUMN    | csm_switch_inventory_history.type is 'The category of this piece of hardware inventory. For example: "FAN", "PS", "SYSTEM", or "MGMT".';
| COMMENT ON COLUMN    |csm_switch_inventory_history.fw_version is 'The firmware version on this piece of inventory.';

# How to Test
1. Run regression against the new changes.
2. Additional testing from previous version to ensure the DB migrates to 18.0 with all other version upgrades along the way.
example running from versions
15.0 (schema version (ga_release)) which include updates from 15.1, 16.0, 16.1, 16.2, 17.0 and then -----> 18.0
15.1 (schema version) which include updates from 16.0, 16.1, 16.2, 17.0 and then -----> 18.0
16.0 (schema version) which include updates from 16.1, 16.2, 17.0 and then -----> 18.0
16.1 (schema version) which include updates from 16.2, 17.0 and then -----> 18.0
16.2 (schema version) which include updates from 17.0 and then -----> 18.0
17.0 (schema version) -----> 18.0

# Supporting Documentation
Migration script 18.0 (Read the docs section)
Additional Function and Triggers chart in the (csm_db_appendix.rst page)

## To Do
_Make sure you attempted to do the following:_
- [ ] @pdlun92 (check regression)
- [x] @NickyDaB (code review)
- [ ] @fpizzano  (review)
- [x] @mew2057  (update csm_allocation API with core_blink field option/ test/ and review)
